### PR TITLE
:sparkles: [#33] Add reset_db_save_after option to config model

### LINKED ISSUE #33: Ability to configure the duration before logging settings are reverted
From LV 125, discussed with Laurens

The current RESET_DB_SAVE_AFTER configuration is 60 minutes, which is quite short if you want to investigate issues which don't happen regularly.

The request is to extend this default value to either 1 week or to make this duration configurable in the admin in order to provide sufficient insight into issues

### PR DESCRIPTION
**Pull request by stevenbal**
closes #33

### CODE REVIEW

File: log_outgoing_requests/config_reset.py [hunk 1]
```diff
@@ -4,8 +4,10 @@
 from .tasks import reset_config
 
 
-def schedule_config_reset():
-    reset_after = settings.LOG_OUTGOING_REQUESTS_RESET_DB_SAVE_AFTER
+def schedule_config_reset(config):
```
Comment (stevenbal) [inline log_outgoing_requests/config_reset.py:7]: not sure how I can add the `OutgoingRequestsLogConfig` typehint here without the docs check failing, I tried it with `if TYPE_CHECKING` but that didn't work

**Comment by alextreme** [comment]
Ziet er goed uit, wel graag een review van Sergei afwachten voor de zekerheid

**Review by alextreme — APPROVED** [review]

File: log_outgoing_requests/config_reset.py (hunk 1 above)
Comment (sergei-maertens) [reply log_outgoing_requests/config_reset.py:7]: `from __future__ import annotations` and then a conditional import tends to work usually, but instead, why not pass the `self.reset_db_save_after` so you can provide a typehint `reset_after: int | None` so you're only working in primitives :)

File: log_outgoing_requests/models.py [hunk 2]
```diff
@@ -206,13 +206,23 @@ class OutgoingRequestsLogConfig(SingletonModel):
...
+    reset_db_save_after = models.PositiveIntegerField(
+        _("Reset saving logs in database after"),
+        null=True,
+        blank=True,
+        help_text=_(
+            "If configured, after the config has been updated, reset the database logging "
+            "after the specified number of minutes. Note: this overrides the "
+            "LOG_OUTGOING_REQUESTS_RESET_DB_SAVE_AFTER environment variable."
+        ),
+    )
```
Comment (sergei-maertens) [inline log_outgoing_requests/models.py:218]: I'd also add a min value validator to set at least 1 minute, 0 minutes would immediately reset it and makes it unclear in the code if the value is set to `0` or `None` with the soft-boolean check `if config.reset_db_save_after`

Adding the validator removes ambiguity

**Review by sergei-maertens — COMMENTED** [review]
There's also a disclaimer that you can only schedule tasks in the future as long as your broker remembers the tasks.

E.g. if you use Redis and you set the reset to 1 week, it's very likely that the key for that task has expired in redis before that moment arrives, so it will never be reset.

File: log_outgoing_requests/models.py [hunk 3]
```diff
@@ -206,13 +206,27 @@ class OutgoingRequestsLogConfig(SingletonModel):
...
+        null=True,
+        blank=True,
+        validators=[MinValueValidator(1)],
+        help_text=_(
+            "If configured, after the config has been updated, reset the database logging "
+            "after the specified number of minutes. Note: this overrides the "
+            "LOG_OUTGOING_REQUESTS_RESET_DB_SAVE_AFTER environment variable. Additionally, "
+            "depending on the broker that is used, if this value is duration is too long "
+            "the key for the reset task might have expired before that time. So make sure not to "
+            "set too large a value for the reset."
```
Comment (sergei-maertens) [inline log_outgoing_requests/models.py:220]:
```suggestion
            "depending on the broker that is used, if this duration is too long "
            "the key for the reset task might have expired before that time. So make sure not to "
            "set too large a value for the reset."
```

File: tests/test_tasks.py [hunk 4]
```diff
@@ -94,10 +94,41 @@ def test_saving_config_schedules_config_reset(mocker):
...
+    settings, mocker
+):
+    settings.LOG_OUTGOING_REQUESTS_RESET_DB_SAVE_AFTER = 1
+    config = OutgoingRequestsLogConfig.get_solo()
+    config.reset_db_save_after = 2
+    mock_task = mocker.patch(
+        "log_outgoing_requests.config_reset.reset_config.apply_async"
+    )
+    config.reset_db_save_after = 4
+    config.save()
```
Comment (sergei-maertens) [inline tests/test_tasks.py:133]:
```suggestion
    mock_task = mocker.patch(
        "log_outgoing_requests.config_reset.reset_config.apply_async"
    )
    config.reset_db_save_after = 4
    config.save()
```
not sure what the point of that first assignment is :)

**Review by sergei-maertens — APPROVED** [review]
couple more things to polish but otherwise seems good to merge

File: tests/test_tasks.py (hunk 4 above)
Comment (stevenbal) [reply tests/test_tasks.py:133]: hmm no idea why that was there

**Pull request merged**